src/forms/DepositForm.vue

```diff
@@ -296,26 +303,36 @@ const init = async (): Promise<void> => {
   isInitializing.value = true
 
   try {
-    allowances[CURRENCIES.stEth] = await fetchAllowanceByCurrency(
-      CURRENCIES.stEth,
+    allowances[CURRENCIES.depositToken] = await fetchAllowanceByCurrency(
+      CURRENCIES.depositToken,
     )
   } catch (error) {
     emit('cancel')
     ErrorHandler.process(error)
   }
 
+  form.lockPeriod = String(userPoolData.value?.claimLockEnd?.toNumber() || '')
+
   isInitializing.value = false
 }
 
 watch(
   () => [
     props.poolId,
-    web3ProvidersStore.erc1967ProxyContract.providerBased.value.address,
     web3ProvidersStore.provider.selectedAddress,
     form.lockPeriod,
+    userPoolData.value?.claimLockEnd,
   ],
-  () => fetchExpectedMultiplier(form.lockPeriod),
-  { immediate: true },
+  async () => {
+    if (!form.lockPeriod) {
+      form.lockPeriod = String(
+        userPoolData.value?.claimLockEnd?.toNumber() || '',
+      )
+    }
+    if (isMultiplierShown.value) {
+      await fetchExpectedMultiplier(form.lockPeriod)
+    }
+  },
```

Comment:
Seems like it should be 2 different `watch` hooks

Reply:
In order to `fetchExpectedMultiplier` logic I also should watch to 

```
props.poolId,
web3ProvidersStore.provider.selectedAddress,
userPoolData.value?.claimLockEnd,
```

editions

Reply:
but do you required to watch this deps in order to set `form.lockPeriod`?